core/src/main/scala/org/http4s/rho/CompileService.scala

```diff
@@ -43,4 +43,20 @@ object CompileService {
     val tree = filter(routes).foldLeft(PathTree()){ (t, r) => t.appendRoute(r) }
     Service.lift { req => tree.getResult(req).toResponse }
   }
+
+
+  /** Convert the `Seq` of [[RhoRoute]]'s into a `AuthedService`
+    *
+    * @tparam U AuthInfo type for this service
+    * @param routes `Seq` of routes to bundle into a service.
+    * @param filter [[RhoMiddleware]] to apply to the routes.
+    * @param ev [[AuthInfo]] instance for U
+    * @return An `AuthedService`
+    */
+  def foldAuthedServices[U](routes: Seq[RhoRoute.Tpe], filter: RhoMiddleware = identity)(implicit ev: AuthInfo[U]): AuthedService[U] = {
```

Comment:
Why can't this live in `AuthedServiceBuilder`? Furthermore, it seems like it's only purpose is to add the entry to the attribute map so couldn't it be
```scala
val rhoService = foldServices(routes, filter)
Service.lift { case AuthedRequest(auth, req) =>
  rhoService(req.withAttribute ...)
}
```